pandas/core/reshape/melt.py

```diff
@@ -39,6 +39,16 @@ def melt(
     else:
         cols = list(frame.columns)
 
+    if 'variable' in frame.columns:
+        warnings.warn(
+            "Using column name 'variable' might cause "
```

Comment:
I think instead we should mention that a `ValueError` will raise if `frame.columns.name is None` in the future.

So this warning should be placed below by:

```
            var_name = [
                frame.columns.name if frame.columns.name is not None else "variable"
            ]
```